app/controllers/services_controller.rb

```diff
@@ -13,20 +13,32 @@ class ServicesController < ApplicationController
 
   def index
     # use SFSG as default site
-    # TODO: be able to parse both categories and eligibilities at once
-    if params[:category_id]
-      all_services = find_by_category(params[:category_id], @site_id_string)
-    elsif params[:eligibility_id]
-      all_services = find_by_eligibility(params[:eligibility_id], @site_id_string)
-    end
+    matching_services = find_services(
+      params[:category_id],
+      params[:eligibility_id],
+      @site_id_string
+    )
 
-    render json: ServicesWithResourceLitePresenter.present(all_services)
+    render json: ServicesWithResourceLitePresenter.present(matching_services)
   end
 
   def read_site_id_string
     @site_id_string = params[:site_id] || Site.find_by(site_code: 'sfsg').id.to_s
   end
 
+  def find_services(category_id, eligibility_id, site_id)
+    if category_id == 'all' || eligibility_id == 'all'
```

Comment:
Wouldn't it be preferable to assume that you are looking for all categories/eligibilities if only site id is specified? I think the default assumption is that only the criteria explicitly passed in to the query string are used to filter results.

Reply:
Updated